datasets/daily_dialog/daily_dialog.py

```diff
@@ -0,0 +1,157 @@
+        # download and extract URLs
+        dl_dir = dl_manager.download_and_extract(_URL)
+        data_dir = os.path.join(dl_dir, "ijcnlp_dailydialog")
+
+        # The splits are nested inside the zip
+        for name in ("train", "validation", "test"):
+            zip_fpath = os.path.join(data_dir, f"{name}.zip")
+            with ZipFile(zip_fpath) as zip_file:
+                zip_file.extractall(path=data_dir)
+                zip_file.close()
+
+        return [
+            nlp.SplitGenerator(
+                name=nlp.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "file_path": os.path.join(data_dir, "train", "dialogues_train.txt"),
+                    "act_path": os.path.join(data_dir, "train", "dialogues_act_train.txt"),
+                    "emotion_path": os.path.join(data_dir, "train", "dialogues_emotion_train.txt"),
+                    "split": "train",
+                },
+            ),
+            nlp.SplitGenerator(
+                name=nlp.Split.TEST,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "file_path": os.path.join(data_dir, "test", "dialogues_test.txt"),
+                    "act_path": os.path.join(data_dir, "test", "dialogues_act_test.txt"),
+                    "emotion_path": os.path.join(data_dir, "test", "dialogues_emotion_test.txt"),
+                    "split": "test",
+                },
+            ),
+            nlp.SplitGenerator(
+                name=nlp.Split.VALIDATION,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "file_path": os.path.join(data_dir, "validation", "dialogues_validation.txt"),
+                    "act_path": os.path.join(data_dir, "validation", "dialogues_act_validation.txt"),
+                    "emotion_path": os.path.join(data_dir, "validation", "dialogues_emotion_validation.txt"),
+                    "split": "dev",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, file_path, act_path, emotion_path, split):
+        """ Yields examples. """
+        # Yields (key, example) tuples from the dataset
+        with open(file_path, "r", encoding="utf-8") as f, open(act_path, "r", encoding="utf-8") as act, open(
+            emotion_path, "r", encoding="utf-8"
+        ) as emotion:
+            for i, (line_f, line_act, line_emotion) in enumerate(zip(f, act, emotion)):
+                if len(line_f.strip()) == 0:
+                    break
+                dialog = line_f.split(self.__EOU__)[:-1]
+                act = line_act.split(" ")[:-1]
+                emotion = line_emotion.split(" ")[:-1]
+
+                assert len(dialog) == len(act) == len(emotion), "Different turns btw dialogue & emotion & action"
+
+                yield f"{split}-i", {
```

Comment:
```suggestion
                yield f"{split}-{i}", {
```

Reply:
Good catch 😱

Looks like there's no check of the unicity of the generated index?

Reply:
We don't use the example ids in practice. This is (yet) another thing that comes from TFDS